### DISCUSSION TITLE
Script To Automate The Import/Export Of Netbox Users And Permissions Between Versions

### DISCUSSION THREAD

**Discussion by mr1716** [netbox-community/netbox#15884] [Ideas]
Pardon if this is already handled during the upgrade when backing up and then restoring the PostGreSQL backup when upgrading. But thought it would be great if there was a way to have Netbox import/export users, groups, and permissions between versions. Like it would be something (a script) that would save user, group, and permission information, before the update, and then be run after the update to restore that information.

**Comment by candlerb** [comment]
The permissions are all in the postgres database. The update applies incremental changes to the database schema, generally without losing any data, so these are all preserved.

However, the postgres database *isn't* backed up before the upgrade, but you would be well advised to do so in case the upgrade leaves your database in a corrupt state - instructions [here](https://docs.netbox.dev/en/stable/administration/replicating-netbox/).
